data/js/skill/mining.js

```diff
@@ -194,6 +213,12 @@ function onCallback1( socket, ourObj )
 				}
 				else if( staticTile == false ) // tileID is from a map tile
 				{
+					if(validMapSandTilesIDs.indexOf(tileID) != -1 && pUser.GetTag( "GatheringSand" ) == 1)
+					{
+						validTileIDFound = true;
+						SandMining( socket, mChar, targX, targY );
+					}
+
 					if( validMapMountainIDs.indexOf( tileID ) != -1 || validMapCaveFloorIDs.indexOf( tileID ) != -1 ) // Mountains or Cave Floors
```

Comment:
Turn this into an "else if" - no need to check if tileID is a valid mountain/cave floor tile if we've already verified that it's a sand tile.